src/test/kotlin/uk/gov/dluhc/notificationsapi/mapper/BespokeCommTemplatePreviewDtoMapperTest.kt

```diff
@@ -0,0 +1,117 @@
+package uk.gov.dluhc.notificationsapi.mapper
+
+import org.assertj.core.api.Assertions
+import org.junit.jupiter.api.extension.ExtendWith
+import org.mockito.InjectMocks
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.kotlin.given
+import org.mockito.kotlin.verify
+import uk.gov.dluhc.notificationsapi.dto.LanguageDto
+import uk.gov.dluhc.notificationsapi.dto.NotificationType
+import uk.gov.dluhc.notificationsapi.models.CommunicationChannel
+import uk.gov.dluhc.notificationsapi.models.Language
+import uk.gov.dluhc.notificationsapi.models.SourceType
+import uk.gov.dluhc.notificationsapi.testsupport.annotations.CommunicationChannelsTest
+import uk.gov.dluhc.notificationsapi.testsupport.testdata.api.buildBespokeCommPersonalisation
+import uk.gov.dluhc.notificationsapi.testsupport.testdata.api.buildGenerateBespokeCommTemplatePreviewRequest
+import uk.gov.dluhc.notificationsapi.testsupport.testdata.dto.buildAddressDto
+import uk.gov.dluhc.notificationsapi.testsupport.testdata.dto.buildBespokeCommPersonalisationDto
+import uk.gov.dluhc.notificationsapi.testsupport.testdata.dto.buildBespokeCommTemplatePreviewDto
+import uk.gov.dluhc.notificationsapi.testsupport.testdata.dto.buildContactDetailsDto
+import java.time.LocalDate
+
+@ExtendWith(MockitoExtension::class)
+class BespokeCommTemplatePreviewDtoMapperTest {
+
+    @Mock
+    private lateinit var languageMapper: LanguageMapper
+
+    @Mock
+    private lateinit var sourceTypeMapper: SourceTypeMapper
+
+    @Mock
+    private lateinit var communicationChannelMapper: CommunicationChannelMapper
+
+    @Mock
+    private lateinit var deadlineMapper: DeadlineMapper
+
+    @InjectMocks
+    private lateinit var mapper: BespokeCommTemplatePreviewDtoMapperImpl
+
+    @CommunicationChannelsTest
+    fun `should map bespoke comm template preview request to dto`(channel: CommunicationChannel) {
+        // Given
+        val subjectHeader = "Your postal vote"
+        val details = "Postal vote details"
+        val whatToDo = "What you need to do"
+        val deadlineDate = LocalDate.of(2024, 7, 10)
+        val deadlineTime = "17:00"
+
+        val request = buildGenerateBespokeCommTemplatePreviewRequest(
+            channel = channel,
+            personalisation = buildBespokeCommPersonalisation(
+                subjectHeader = subjectHeader,
+                details = details,
+                whatToDo = whatToDo,
+                deadlineDate = deadlineDate,
+                deadlineTime = deadlineTime,
+            ),
+        )
+        val expectedChannel = uk.gov.dluhc.notificationsapi.dto.CommunicationChannel.valueOf(channel.name)
```

Comment:
could use a named import `import uk.gov.dluhc.notificationsapi.dto.CommunicationChannel as CommunicationChannelDto` to avoid naming clash and slightly neaten code

similar comment for `SourceType`